include/effectmanager.h

```diff
@@ -595,25 +595,37 @@ class  EffectManager : public IJSONSerializable
     {
         // If the Interval is set to zero, we treat that as an infinite interval and don't even look at the time used so far
         uint timeUsedByCurrentEffect = GetTimeUsedByCurrentEffect();
-        uint interval = GetInterval();
+        uint interval = GetEffectiveInterval();
 
         return timeUsedByCurrentEffect > interval ? 0 : (interval - timeUsedByCurrentEffect);
     }
 
-    uint GetInterval() const
+    uint GetEffectiveInterval() const
     {
+        auto& currentEffect = GetCurrentEffect();
         // This allows you to return a MaximumEffectTime and your effect won't be shown longer than that
-        return min((_effectInterval == 0 ? std::numeric_limits<uint>::max() : _effectInterval), GetCurrentEffect().MaximumEffectTime());
+        return min((IsIntervalEternal() ? std::numeric_limits<uint>::max() : _effectInterval),

```

Comment:
I thought about that, and decided against it here. 0 is a nonsensical value in both cases, so needs to be special-cased anyway. Because of that I concluded it made more sense to make 0 equal to absent, and provide the "Has/Is" functions to not make the class's users have to remember that. Otherwise, we'd have "absent", "nonsense", "some value". A secondary argument is that we do not have an "OptionalBigPositiveInteger" data type towards the front-end yet, and I'm not too eager to add one.